frontend/packages/dev-console/src/components/pipelines/pipeline-builder/utils.ts

```diff
@@ -33,11 +33,12 @@ export const convertResourceToTask = (
   resource: PipelineResourceTask,
   runAfter?: string[],
 ): PipelineTask => {
+  const kind = resource.kind ?? TaskModel.kind;
   return {
-    name: resource.metadata.name,
+    name: `${resource.metadata.name}-${kind.toLowerCase()}`,
```

Comment:
If we do this, we are not really avoiding the problem entirely. We are likely to be okay, but let's avoid anything that will lead to potentially another bug (or if QE is on game, a failure of the QE Review).

Recommend a generate name function (maybe we can put this in shared, or here -- wherever for now):

```
export const safeName = (reservedNames: string[], desiredName: string): string => {
  if (reservedNames.includes(desiredName)) {
    const newName = `${desiredName}-${getRandomChars()}`; // maybe `getRandomChars(3)`?
    if (reservedNames.includes(newName)) {
      return safeName(reservedNames, desiredName);
    }
    return newName;
  }
  return desiredName;
};

// use like:
name: safeName(reservedNames, resource.metadata.name),
```

Then you update the two uses of `convertResourceToTask` to include a 3rd param (I'd make it the first param personally) of a `string[]`. 

Sample usage:
```
  const usedNames = tasks.map((t) => t.name);
  const newPipelineTask: PipelineTask = convertResourceToTask(usedNames, resource, runAfter);
```